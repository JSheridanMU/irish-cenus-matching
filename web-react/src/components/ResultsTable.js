import React from 'react'
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableContainer,
  IconButton,
  Box,
  Collapse,
} from '@material-ui/core'
import FieldImporter from './form-fields/FieldImporter'
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown'
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp'
import Tooltip from '@material-ui/core/Tooltip'
import Comparison from './Comparison'

function formatData(data) {
  let people = []
  data.Person.forEach((person) => {
    people.push({
      age: person.age,
      birthplace: person.birthplace,
      forename: person.forename,
      household: person.household,
      occupation: person.occupation,
      relationToHead: person.relationToHead,
      religion: person.religion,
      sex: person.sex,
      surname: person.surname,
      relatives: person.related_to.concat(person.related_from),
      relationships: person.RELATED_TO_rel.from.concat(
        person.RELATED_TO_rel.to
      ),
      id: person.id,
    })
  })
  return people
}

function Row(props) {
  const { row, ...other } = props
  const [open, setOpen] = React.useState(false)

  const trigger = () => {
    other.searchTrigger(row)
  }

  return (
    <React.Fragment>
      <TableRow>
        <TableCell component="th" scope="row">
          {row.forename}
        </TableCell>
        <TableCell>{row.surname}</TableCell>
        <TableCell>{row.age}</TableCell>
        <TableCell>{row.sex}</TableCell>
        <TableCell>{row.birthplace}</TableCell>
        <TableCell>{row.occupation}</TableCell>
        <TableCell>{row.religion}</TableCell>
        <TableCell>{row.relationToHead}</TableCell>
        <Tooltip
          title="View the original household return form"
          placement="top"
        >
          <TableCell>
            <FieldImporter.Button
              text="Household"
              color="primary"
              onClick={(e) => {
                e.preventDefault()
                window.open(row.household, '_blank')
              }}
            />
          </TableCell>
        </Tooltip>
        <TableCell>
          {!row.relatives.length ? null : (
            <IconButton
              aria-label="expand row"
              size="small"
              onClick={() => setOpen(!open)}
            >
              {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
            </IconButton>
          )}
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={9}>
          <Collapse in={open} timeout="auto" unmountOnExit>
            <Box sx={{ margin: 1 }}>
              <Table size="small">
                <TableHead>
                  <TableRow>
                    <TableCell>Forename</TableCell>
                    <TableCell>Surname</TableCell>
                    <TableCell>Year of Birth</TableCell>
                    <TableCell>Sex</TableCell>
                    <TableCell>Birthplace</TableCell>
                    <TableCell>Occupation</TableCell>
                    <TableCell>Religion</TableCell>
                    <TableCell>Relation to Head</TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      {row.forename}
                    </TableCell>
                    <TableCell>{row.surname}</TableCell>
                    <TableCell>{row.age}</TableCell>
                    <TableCell>{row.sex}</TableCell>
                    <TableCell>{row.birthplace}</TableCell>
                    <TableCell>{row.occupation}</TableCell>
                    <TableCell>{row.religion}</TableCell>
                    <TableCell>{row.relationToHead}</TableCell>
                  </TableRow>
                  {row.relatives.map((relative) => (
                    <TableRow key={relative.id}>
                      <TableCell component="th" scope="row">
                        {relative.forename}
                      </TableCell>
                      <TableCell>{relative.surname}</TableCell>
                      <TableCell>{relative.age}</TableCell>
                      <TableCell>{relative.sex}</TableCell>
                      <TableCell>{relative.birthplace}</TableCell>
                      <TableCell>{relative.occupation}</TableCell>
                      <TableCell>{relative.religion}</TableCell>
                      <TableCell>{relative.relationToHead}</TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
              {other.firstSearch ? (
                <Tooltip
                  title={
                    row.household.includes('1911')
                      ? 'Trigger a search of this household in 1901'
                      : 'Trigger a search of this household in 1911'
                  }
                  placement="right"
                >
                  <FieldImporter.Button
                    text={
                      row.household.includes('1911')
                        ? 'Search 1901'
                        : 'Search 1911'
                    }
                    color="primary"
                    onClick={trigger}
                  />
                </Tooltip>
              ) : null}
              {other.secondSearch ? (
                <Tooltip
                  title="Open a graph visualisation of both households for comparisson"
                  placement="right"
                >
                  <Comparison original={other.query} new={row} />
                </Tooltip>
              ) : null}
            </Box>
          </Collapse>
        </TableCell>
      </TableRow>
    </React.Fragment>
  )
}

export default function ResultsTable(props) {
  const { data, loading, error, ...other } = props

  return data ? (
    data.Person.length === 0 ? (
      <React.Fragment>No Results</React.Fragment>
    ) : (
      <React.Fragment>
        {data && !loading && !error && (
          <TableContainer>
            <Table stickyHeader size="small">
              <TableHead>
                <TableRow>
                  <TableCell>Forename</TableCell>
                  <TableCell>Surname</TableCell>
                  <Tooltip
                    title="Year of Birth is estimated with age and year"
                    placement="top"
                  >
                    <TableCell>Year of Birth</TableCell>
                  </Tooltip>
                  <TableCell>Sex</TableCell>
                  <TableCell>Birthplace</TableCell>
                  <TableCell>Occupation</TableCell>
                  <TableCell>Religion</TableCell>
                  <TableCell>Relation to Head</TableCell>
                  <TableCell>Household</TableCell>
                  <TableCell />
                </TableRow>
              </TableHead>
              <TableBody>
                {formatData(props.data).map((row) => (
                  <Row key={row.id} row={row} {...other} />
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        )}
      </React.Fragment>
    )
  ) : null
}
