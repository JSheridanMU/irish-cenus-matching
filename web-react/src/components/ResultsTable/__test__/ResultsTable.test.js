import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ResultsTable from '.././ResultsTable'

afterEach(cleanup)

const firstProps = {
  data: {
    Person: [
      {
        __typename: 'Person',
        id: '69701',
        forename: 'James',
        surname: 'Sheridan',
        age: 1881,
        sex: 'M',
        relationToHead: 'Nephew',
        household:
          'http://www.census.nationalarchives.ie/pages/1911/Kildare/Leixlip/Leixlip_Town/540364/',
        birthplace: 'Co Kildare',
        occupation: 'Postman',
        religion: 'Roman Catholic',
        soundex: 'J520S635',
        hisco: null,
        related_to: [
          {
            __typename: 'Person',
            id: '59687',
            forename: 'Catherine',
            surname: 'Sheridan',
            age: 1847,
            sex: 'F',
            relationToHead: 'Wife',
            birthplace: 'Co Kildare',
            occupation: 'UNKNOWN',
            religion: 'Roman Catholic',
            soundex: 'K365S635',
            hisco: null,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J520S635-K365S635-34',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-K365S635-0',
                },
              ],
              to: [],
            },
          },
        ],
        related_from: [
          {
            __typename: 'Person',
            id: '79799',
            forename: 'John',
            surname: 'Sheridan',
            age: 1847,
            sex: 'M',
            relationToHead: 'Head of Family',
            birthplace: 'Co Kildare',
            occupation: 'Master Bootmaker',
            religion: 'Roman Catholic',
            soundex: 'J500S635',
            hisco: null,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [],
              to: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-K365S635-0',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-J520S635-34',
                },
              ],
            },
          },
        ],
        RELATED_TO_rel: {
          __typename: '_PersonRELATED_TO_relDirections',
          from: [
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J500S635-J520S635-34',
            },
          ],
          to: [
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J520S635-K365S635-34',
            },
          ],
        },
      },
      {
        __typename: 'Person',
        id: '84251',
        forename: 'James',
        surname: 'Sheridan',
        age: 1885,
        sex: 'M',
        relationToHead: 'Son',
        household:
          'http://www.census.nationalarchives.ie/pages/1911/Kildare/Naas_Urban/New_Row/545683/',
        birthplace: 'Co Kildare',
        occupation: 'Groom',
        religion: 'Roman Catholic',
        soundex: 'J520S635',
        hisco: 62490,
        related_to: [
          {
            __typename: 'Person',
            id: '84252',
            forename: 'Patrick',
            surname: 'Sheridan',
            age: 1886,
            sex: 'M',
            relationToHead: 'Son',
            birthplace: 'Co Kildare',
            occupation: 'Labourer',
            religion: 'Roman Catholic',
            soundex: 'P362S635',
            hisco: 99910,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J520S635-P362S635-1',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-P362S635-24',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-P362S635-3',
                },
              ],
              to: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'P362S635-M650S635-13',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'P362S635-J500S635-10',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'P362S635-B632S635-25',
                },
              ],
            },
          },
          {
            __typename: 'Person',
            id: '74488',
            forename: 'Maryann',
            surname: 'Sheridan',
            age: 1899,
            sex: 'F',
            relationToHead: 'Daughter',
            birthplace: 'Co Kildare',
            occupation: 'Scolar',
            religion: 'Roman Catholic',
            soundex: 'M650S635',
            hisco: null,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'P362S635-M650S635-13',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J520S635-M650S635-14',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-M650S635-10',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-M650S635-11',
                },
              ],
              to: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M650S635-B632S635-38',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M650S635-J500S635-3',
                },
              ],
            },
          },
          {
            __typename: 'Person',
            id: '67895',
            forename: 'Bridget',
            surname: 'Sheridan',
            age: 1861,
            sex: 'F',
            relationToHead: 'Head of Family',
            birthplace: 'Co Kildare',
            occupation: 'UNKNOWN',
            religion: 'Roman Catholic',
            soundex: 'B632S635',
            hisco: null,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-B632S635-35',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M650S635-B632S635-38',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-B632S635-28',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-B632S635-49',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'P362S635-B632S635-25',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J520S635-B632S635-24',
                },
              ],
              to: [],
            },
          },
          {
            __typename: 'Person',
            id: '67896',
            forename: 'John',
            surname: 'Sheridan',
            age: 1896,
            sex: 'M',
            relationToHead: 'Son',
            birthplace: 'Co Kildare',
            occupation: 'Groom',
            religion: 'Roman Catholic',
            soundex: 'J500S635',
            hisco: 62490,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'P362S635-J500S635-10',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M650S635-J500S635-3',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-J500S635-14',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-J500S635-7',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J520S635-J500S635-11',
                },
              ],
              to: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-B632S635-35',
                },
              ],
            },
          },
        ],
        related_from: [
          {
            __typename: 'Person',
            id: '95219',
            forename: 'Mary Bridget',
            surname: 'Sheridan',
            age: 1910,
            sex: 'F',
            relationToHead: 'Daughter',
            birthplace: 'Co Kildare',
            occupation: 'UNKNOWN',
            religion: 'Roman Catholic',
            soundex: 'M616S635',
            hisco: null,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-M616S635-21',
                },
              ],
              to: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-P362S635-24',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-J520S635-25',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-M650S635-11',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-B632S635-49',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'M616S635-J500S635-14',
                },
              ],
            },
          },
          {
            __typename: 'Person',
            id: '95218',
            forename: 'Jane',
            surname: 'Sheridan',
            age: 1889,
            sex: 'F',
            relationToHead: 'Daughter',
            birthplace: 'Co Kildare',
            occupation: 'UNKNOWN',
            religion: 'Roman Catholic',
            soundex: 'J500S635',
            hisco: null,
            RELATED_TO_rel: {
              __typename: '_PersonRELATED_TO_relDirections',
              from: [],
              to: [
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-M650S635-10',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-M616S635-21',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-J520S635-4',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-P362S635-3',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-B632S635-28',
                },
                {
                  __typename: '_PersonRELATED_TO_rel',
                  name: 'J500S635-J500S635-7',
                },
              ],
            },
          },
        ],
        RELATED_TO_rel: {
          __typename: '_PersonRELATED_TO_relDirections',
          from: [
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'M616S635-J520S635-25',
            },
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J500S635-J520S635-4',
            },
          ],
          to: [
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J520S635-P362S635-1',
            },
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J520S635-M650S635-14',
            },
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J520S635-B632S635-24',
            },
            {
              __typename: '_PersonRELATED_TO_rel',
              name: 'J520S635-J500S635-11',
            },
          ],
        },
      },
    ],
  },
  loading: false,
  firstSearch: false,
}

test('renders without crashing', () => {
  render(<ResultsTable data={firstProps.data} firstSearch={true} />)
})

test('renders content of search', () => {
  const { getAllByText } = render(
    <ResultsTable data={firstProps.data} firstSearch={true} />
  )

  getAllByText('James')
  getAllByText('Sheridan')
  getAllByText('1881')
  getAllByText('M')
  getAllByText('Co Kildare')
  getAllByText('Postman')
  getAllByText('Roman Catholic')
  getAllByText('Nephew')
})

test('test dropdown', () => {
  const { getAllByText, getByTestId } = render(
    <ResultsTable data={firstProps.data} firstSearch={true} />
  )

  const button = getByTestId('dropdown')
  fireEvent.click(button)
  getAllByText('Search 1901')
})

test('matches snapshot', () => {
  const tree = renderer
    .create(<ResultsTable data={firstProps.data} firstSearch={true} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
