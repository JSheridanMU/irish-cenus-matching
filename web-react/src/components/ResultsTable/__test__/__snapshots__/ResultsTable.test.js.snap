// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root MuiTable-stickyHeader"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Forename
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Surname
        </th>
        <th
          aria-describedby={null}
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          scope="col"
          title="Year of Birth is estimated with age and year"
        >
          Birthyear
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Sex
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Birthplace
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Occupation
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Religion
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Relation to Head
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        >
          Household
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          role="cell"
          scope="row"
        >
          James
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Sheridan
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          1881
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          M
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Co Kildare
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Postman
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Roman Catholic
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Nephew
        </td>
        <td
          aria-describedby={null}
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="View the original household return form"
        >
          <button
            className="MuiButtonBase-root MuiButton-root makeStyles-root-8 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label makeStyles-label-9"
            >
              Household
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <button
            aria-label="expand row"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            data-testid="dropdown"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          colSpan={9}
          style={
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        />
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          role="cell"
          scope="row"
        >
          James
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Sheridan
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          1885
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          M
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Co Kildare
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Groom
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Roman Catholic
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Son
        </td>
        <td
          aria-describedby={null}
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="View the original household return form"
        >
          <button
            className="MuiButtonBase-root MuiButton-root makeStyles-root-8 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label makeStyles-label-9"
            >
              Household
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <button
            aria-label="expand row"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            data-testid="ignore"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          colSpan={9}
          style={
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;
